core/ts/Poll.ts

```diff
@@ -121,33 +122,29 @@ export class Poll implements IPoll {
     maciStateRef: MaciState,
     stateTreeDepth: number,
   ) {
-    this.duration = duration;
     this.pollEndTimestamp = pollEndTimestamp;
     this.coordinatorKeypair = coordinatorKeypair;
     this.treeDepths = treeDepths;
     this.batchSizes = batchSizes;
     this.maxValues = maxValues;
     this.maciStateRef = maciStateRef;
     this.pollId = maciStateRef.polls.length;
-    this.numSignUps = Number(maciStateRef.numSignUps.toString());
     this.stateTreeDepth = stateTreeDepth;
 
-    this.stateTree = new IncrementalQuinTree(this.stateTreeDepth, blankStateLeafHash, STATE_TREE_ARITY, hash5);
     this.messageTree = new IncrementalQuinTree(
       this.treeDepths.messageTreeDepth,
       NOTHING_UP_MY_SLEEVE,
       MESSAGE_TREE_ARITY,
       hash5,
     );
 
-    for (let i = 0; i < this.maxValues.maxVoteOptions; i++) {
-      this.tallyResult.push(BigInt(0));
-      this.perVOSpentVoiceCredits.push(BigInt(0));
-      this.subsidy.push(BigInt(0));
-    }
+    this.tallyResult = new Array(this.maxValues.maxVoteOptions).fill(BigInt(0));
+    this.perVOSpentVoiceCredits = new Array(this.maxValues.maxVoteOptions).fill(BigInt(0));
+    this.subsidy = new Array(this.maxValues.maxVoteOptions).fill(BigInt(0));
 
-    const blankBallot = Ballot.genBlankBallot(this.maxValues.maxVoteOptions, treeDepths.voteOptionTreeDepth);
-    this.ballots.push(blankBallot);
+    // we put a blank state leaf to prevent a DoS attack
```

Comment:
I don't understand how this prevents a DoS attack 🤔

Reply:
https://maci.pse.dev/docs/spec -> section 6.1 

It's explained in more details there :)

Reply:
let's add it in the code comments too?

Reply:
Nice, thanks